src/core/config/openthread-core-config-check.h

```diff
@@ -518,6 +518,24 @@
 
 #endif // OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
 
+#if OPENTHREAD_CONFIG_SRP_SERVER_ENABLE
+#if !OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE
+#error "OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE is required for OPENTHREAD_CONFIG_SRP_SERVER_ENABLE"
+#endif
+
+#if !OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE
+#error "OPENTHREAD_CONFIG_TMF_NETDATA_SERVICE_ENABLE is required for OPENTHREAD_CONFIG_SRP_SERVER_ENABLE"
+#endif
+
+#if !OPENTHREAD_CONFIG_ECDSA_ENABLE
+#error "OPENTHREAD_CONFIG_ECDSA_ENABLE is required for OPENTHREAD_CONFIG_SRP_SERVER_ENABLE"
+#endif
+
+#if !OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE
+#error "OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE is required for OPENTHREAD_CONFIG_SRP_SERVER_ENABLE"
```

Comment:
I know we are using heap for items in SRP server, but do we need external heap for it? 
I think we can support it with internal heap as well. Maybe we need  a check on heap size?

Reply:
Agree that requiring the external heap is too strict, will remove it.